test/unit/ethereum/Authentication.test.js

```diff
@@ -0,0 +1,40 @@
+const { wait, waitForCondition } = require('streamr-test-utils')
+
+const ethereumAuthenticate = require('../../../src/helpers/ethereumAuthenticate')
+
+// Dev env 1 privateKey and address
+const privateKey = '0xaa7a3b3bb9b4a662e756e978ad8c6464412e7eef1b871f19e5120d4747bce966'
+const address = '0xde1112f631486CfC759A50196853011528bC5FA0'
+
+describe('Ethereum authentication', () => {
+    it('authenticates with a private key', async (done) => {
+        const config = {
+            privateKey
+        }
+        const wallet = await ethereumAuthenticate.authenticateFromConfig(config)
+        await waitForCondition(() => wallet)
+        if (wallet.address === address) {
+            done()
+        }
+    })
+    it('authenticates with a randomly generated wallet', async (done) => {
+        const config = {
+            generateWallet: true
+        }
+        const wallet = await ethereumAuthenticate.authenticateFromConfig(config)
+        await waitForCondition(() => wallet)
+        if (wallet !== {}) {
```

Comment:
You can just do direct assertion to avoid having to deal with done (also the error message will be more informative):
```js
expect(wallet.address).toHaveLength(42)
```